File: Java Barcode Scanner ME/app/src/main/java/de/dimedis/mobileentry/backend/response/Versions.java
package de.dimedis.mobileentry.backend.response;

import com.google.gson.annotations.SerializedName;

import java.util.Objects;

public class Versions {
    public static final String LOCAL_CONFIG = "local_config";
    public static final String APP = "app";
    public static final String MY_AVAILABLE_BORDERS = "my_available_borders";
    public static final String LANGUAGES = "languages";
    public static final String LIBRARY = "library";
    public static final String SETTINGS = "settings";


    @SerializedName(LOCAL_CONFIG)
    private String local_config = "1";//stub

    @SerializedName(APP)
    private String app;

    @SerializedName(MY_AVAILABLE_BORDERS)
    private String my_available_borders;

    @SerializedName(LANGUAGES)
    private String languages;

    @SerializedName(LIBRARY)
    private String library;

    @SerializedName(SETTINGS)
    private String settings = "0";

    public String getLocalConfig() {
        return local_config;
    }

    public String getApp() {
        return app;
    }

    public String getMyAvailableBorders() {
        return my_available_borders;
    }

    public String getLanguages() {
        return languages;
    }

    public String getLibrary() {
        return library;
    }

    public String getSettings() {
        return settings;
    }

    public void setLocalConfig(String local_config) {
        this.local_config = local_config;
    }

    public void setApp(String app) {
        this.app = app;
    }

    public void setMyAvailableBorders(String my_available_borders) {
        this.my_available_borders = my_available_borders;
    }

    public void setLanguages(String languages) {
        this.languages = languages;
    }

    public void setLibrary(String library) {
        this.library = library;
    }

    public void setSettings(String settings) {
        this.settings = settings;
    }

    @Override
    public String toString() {
        return "Versions{" +
                "local_config='" + local_config + '\'' +
                ", app='" + app + '\'' +
                ", my_available_borders='" + my_available_borders + '\'' +
                ", languages='" + languages + '\'' +
                ", library='" + library + '\'' +
                ", settings='" + settings + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Versions)) return false;

        Versions versions = (Versions) o;

        if (!Objects.equals(local_config, versions.local_config))
            return false;
        if (!Objects.equals(app, versions.app)) return false;
        if (!Objects.equals(my_available_borders, versions.my_available_borders))
            return false;
        if (!Objects.equals(languages, versions.languages))
            return false;
        if (!Objects.equals(library, versions.library))
            return false;
        return Objects.equals(settings, versions.settings);
    }

    @Override
    public int hashCode() {
        int result = local_config != null ? local_config.hashCode() : 0;
        result = 31 * result + (app != null ? app.hashCode() : 0);
        result = 31 * result + (my_available_borders != null ? my_available_borders.hashCode() : 0);
        result = 31 * result + (languages != null ? languages.hashCode() : 0);
        result = 31 * result + (library != null ? library.hashCode() : 0);
        result = 31 * result + (settings != null ? settings.hashCode() : 0);
        return result;
    }
}

File: Kotlin Migration/NTMobileEntry/app/src/main/java/de/dimedis/mobileentry/backend/response/Versions.kt
package de.dimedis.mobileentry.backend.response

import com.google.gson.annotations.SerializedName

class Versions {
    @SerializedName(LOCAL_CONFIG)
    var localConfig: String? = "1" //stub

    @SerializedName(APP)
    var app: String? = null

    @SerializedName(MY_AVAILABLE_BORDERS)
    var myAvailableBorders: String? = null

    @SerializedName(LANGUAGES)
    var languages: String? = null

    @SerializedName(LIBRARY)
    var library: String? = null

    @SerializedName(SETTINGS)
    var settings: String? = "0"
    override fun toString(): String {
        return "Versions{" +
                "local_config='" + localConfig + '\'' +
                ", app='" + app + '\'' +
                ", my_available_borders='" + myAvailableBorders + '\'' +
                ", languages='" + languages + '\'' +
                ", library='" + library + '\'' +
                ", settings='" + settings + '\'' +
                '}'
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Versions) return false
        if (localConfig != other.localConfig) return false
        if (app != other.app) return false
        if (myAvailableBorders != other.myAvailableBorders) return false
        if (languages != other.languages) return false
        return if (library != other.library) false else settings == other.settings
    }

    override fun hashCode(): Int {
        var result = if (localConfig != null) localConfig.hashCode() else 0
        result = 31 * result + if (app != null) app.hashCode() else 0
        result = 31 * result + if (myAvailableBorders != null) myAvailableBorders.hashCode() else 0
        result = 31 * result + if (languages != null) languages.hashCode() else 0
        result = 31 * result + if (library != null) library.hashCode() else 0
        result = 31 * result + if (settings != null) settings.hashCode() else 0
        return result
    }

    companion object {
        const val LOCAL_CONFIG = "local_config"
        const val APP = "app"
        const val MY_AVAILABLE_BORDERS = "my_available_borders"
        const val LANGUAGES = "languages"
        const val LIBRARY = "library"
        const val SETTINGS = "settings"
    }
}